client/src/index.jsx

```diff
@@ -94,14 +96,27 @@ class App extends React.Component {
       }
     }
   }
+  
+  handleCarouselToggle() {
```

Comment:
I think you can simplify this to
`const { showCarousel } = this.state;
      this.setState({
        showCarousel: !showCarousel,
      });`